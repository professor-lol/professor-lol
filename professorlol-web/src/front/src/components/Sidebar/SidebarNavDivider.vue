<template>
  <li :class="classList" />
</template>

<script>
export default {
  name: 'SidebarNavDivider',
  props: {
    classes: {
      type: [String, Array, Object],
      default: ''
    }
  },
  computed: {
    classList () {
      return [
        'nav-divider',
        ...this.itemClasses
      ]
    },
    itemClasses () {
      const classes = this.classes
      return !classes ? [] : typeof classes === 'string' || classes instanceof String ? classes.split(' ') : Array.isArray(classes) ? classes : Object.keys(classes).filter(i=>classes[i])
    }
  }
}
</script>
