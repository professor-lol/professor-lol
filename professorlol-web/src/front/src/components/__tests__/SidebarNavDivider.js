import { mount } from '@vue/test-utils'
import SidebarNavDivider from "../Sidebar/SidebarNavDivider";

describe("SidebarNavDivider.vue", () => {
  // Inspect the raw component options
  it('has hasSlotDefault computed property', () => {
    expect(typeof SidebarNavDivider.computed.classList).toBe('function')
    expect(typeof SidebarNavDivider.computed.itemClasses).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarNavDivider)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.classes()).toContain('nav-divider')
  })
});
