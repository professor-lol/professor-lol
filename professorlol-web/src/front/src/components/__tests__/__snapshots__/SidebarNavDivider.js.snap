// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavDivider.vue renders correctly 1`] = `
<li
  class="nav-divider"
/>
`;
